VideoApp/VideoApp/Video/VideoAppCameraSource.swift

```diff
@@ -0,0 +1,168 @@
+    }
+
+    private func createCameraSource() {
+        let options = CameraSourceOptions() { builder in
+            if #available(iOS 13, *) {
+                builder.orientationTracker = UserInterfaceTracker(scene: UIApplication.shared.keyWindow!.windowScene!)
+            }
+            
+            switch self.appSettingsStore.topology {
+            case .group:
+                // Take a best guess and remove rotation tags using hardware acceleration
+                builder.rotationTags = .remove
+            case .peerToPeer:
+                break
+            }
+        }
+        
+        cameraSource = CameraSource(options: options, delegate: self)
+        
+        if let cameraSource = cameraSource {
+            localVideoTrack = LocalVideoTrack(source: cameraSource, enabled: true, name: "camera")
+            startCameraSource()
+        }
+    }
+    
+    private func startCameraSource() {
+        guard let cameraSource = cameraSource else { return }
+
+        let targetSize: CMVideoDimensions
+        let cropRatio: CGFloat
+        let frameRate: UInt
+        
+        switch appSettingsStore.videoCodec {
+        case .h264, .vp8:
+            // 640 x 480 squarish crop (1.13:1)
+            targetSize = CMVideoDimensions(width: 544, height: 480)
+            
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 20
+        case .vp8Simulcast:
+            // 1024 x 768 squarish crop (1.25:1) on most iPhones. 1280 x 720 squarish crop (1.25:1) on the iPhone X
+            // and models that don't have 1024 x 768.
+            targetSize = CMVideoDimensions(width: 900, height: 720)
+            
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 24 // With simulcast enabled there are 3 temporal layers, allowing a frame rate of f/4
+        }
+        
+        let captureDevice = CameraSource.captureDevice(position: .front)!
+        let preferredFormat = selectVideoFormatBySize(captureDevice: captureDevice, targetSize: targetSize)
+        preferredFormat.frameRate = min(preferredFormat.frameRate, frameRate)
+        
+        let cropDimensions: CMVideoDimensions
+        
+        if preferredFormat.dimensions.width > preferredFormat.dimensions.height {
+            cropDimensions = CMVideoDimensions(
+                width: Int32(CGFloat(preferredFormat.dimensions.height) * cropRatio),
+                height: preferredFormat.dimensions.height
+            )
+        } else {
+            cropDimensions = CMVideoDimensions(
+                width: preferredFormat.dimensions.width,
+                height: Int32(CGFloat(preferredFormat.dimensions.width) * cropRatio)
+            )
+        }
+        
+        let outputFormat = VideoFormat()
+        outputFormat.dimensions = cropDimensions
+        outputFormat.pixelFormat = preferredFormat.pixelFormat
+        outputFormat.frameRate = 0
```

Comment:
Pretty sure, this is what objc had:
```objc
    TVIVideoFormat *outputFormat = [[TVIVideoFormat alloc] init];	
    outputFormat.dimensions = cropDimensions;	
    outputFormat.pixelFormat = preferredFormat.pixelFormat;	
    outputFormat.frameRate = 0;
```

Reply:
Yes, the *is* what Objc-C had... My question was more around, is `outputFormat.frameRate = 0;` actually correct or something we had set wrong back in the day and it still persisted. 

If you look at `TVIVideoFormat` docs:

```
/**
 *  @brief The frame rate of the video content.
 *
 *  @discussion This value should be greater than zero.
 */
@property (nonatomic, assign) NSUInteger frameRate;
```

Reply:
That does seem like it should not be zero. 😀

But `requestOutputFormat` doc says `If a frame rate is specified then the sink will drop frames to achieve it.` so maybe a little conflicting info. So it seems like zero is appropriate then since this doc is more specific to the situation. Unless we actually want to limit frames...but why wouldn't we do that on the input side if I'm understanding it correctly?

Reply:
At this point let's carry this forward as it is, which is how it was in Obj-C as well.